app/src/evm/apdu_handler_evm.c

```diff
@@ -297,6 +297,7 @@ void handleSignEip191(volatile uint32_t *flags, volatile uint32_t *tx, uint32_t
 
     CHECK_APP_CANARY()
     if (!eip191_msg_parse()) {
```

Comment:
Let's add   *flags |= IO_ASYNCH_REPLY; here or I think this might freeze when showing the error message